lib/new_relic/traced_thread.rb

```diff
@@ -0,0 +1,38 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  #
+  # This class creates a thread that contains instrumentation to allow the agent to see spans created inside of this thread.
+  # In order to have this functionality inserted into all threads automatically,
+  # enable the `instrumentation.thread.tracing` configuration option in your newrelic.yml
```

Comment:
What happens if `instrumentation.thread` is `disabled`? Will this still work?

Reply:
Good point. It will still work, but the span parents will be a little wonky. I'll add something about that config setting as well